File: src/main/java/com/mohe/spring/dto/BookmarkDto.java
package com.mohe.spring.dto;

import jakarta.validation.constraints.NotBlank;
import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.List;

public class BookmarkDto {

    // Bookmark Toggle
    public record BookmarkToggleRequest(
        @NotBlank(message = "장소 ID는 필수입니다")
        String placeId
    ) {}

    public record BookmarkToggleResponse(
        boolean isBookmarked,
        String message
    ) {}

    // Bookmark List
    public record BookmarkListResponse(
        List<BookmarkData> bookmarks
    ) {}

    public record BookmarkData(
        String id,
        BookmarkPlaceData place,
        OffsetDateTime createdAt
    ) {}

    public record BookmarkPlaceData(
        String id,
        String name,
        String location,
        String image,
        BigDecimal rating
    ) {}
}

File: src/main/kotlin/com/mohe/spring/dto/BookmarkDto.kt
package com.mohe.spring.dto

import jakarta.validation.constraints.NotBlank
import java.math.BigDecimal
import java.time.OffsetDateTime

// Bookmark Toggle
data class BookmarkToggleRequest(
    @field:NotBlank(message = "장소 ID는 필수입니다")
    val placeId: String
)

data class BookmarkToggleResponse(
    val isBookmarked: Boolean,
    val message: String
)

// Bookmark List
data class BookmarkListResponse(
    val bookmarks: List<BookmarkData>
)

data class BookmarkData(
    val id: String,
    val place: BookmarkPlaceData,
    val createdAt: OffsetDateTime
)

data class BookmarkPlaceData(
    val id: String,
    val name: String,
    val location: String?,
    val image: String?,
    val rating: BigDecimal
)